packages/components/src/engine/metadata.ts

```diff
@@ -1,12 +1,75 @@
 import type { IsomerPageSchemaType, IsomerSitemap } from "~/types"
 import { getSitemapAsArray } from "~/utils"
 
+const getMetaDescription = (props: IsomerPageSchemaType) => {
+  if (props.meta?.description) {
+    return props.meta.description
+  }
+
+  switch (props.layout) {
+    case "article":
+      return props.page.articlePageHeader.summary
+    case "content":
+    case "database":
+    case "index":
+      return props.page.contentPageHeader.summary
+    case "collection":
+      return props.page.subtitle
+    case "homepage":
+      return props.content.find((item) => item.type === "hero")?.subtitle
+    case "file":
+    case "link":
+    case "search":
+    case "notfound":
+      // NOTE: These pages do not appear in search results, so we don't need to
+      // provide a meta description
+      return undefined
+    default:
+      const _: never = props
+      return undefined
+  }
+}
+
+const getMetaImage = (props: IsomerPageSchemaType) => {
+  switch (props.layout) {
+    case "article":
+      return props.page.image?.src || props.meta?.image
```

Comment:
if `meta.image` is found, which means it's intentionally uploaded by user, should we prioritize that?

similar to `getMetaDescription` where we do
```ts
if (props.meta?.description) {
    return props.meta.description
  }
```

Reply:
Updated in 252ce172e3223a985e958c0b12cc5f2fda117008.